Several methods need await to be properly Async

Articles/Search/Impl/ArticlesDbSearchEngine.cs(22,56): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. 
Pages/CreateArticleFromLink.cshtml.cs(63,36): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
^ those are the two warnings I saw.

--- a/CoreWiki.Application/Articles/Search/Impl/ArticlesDbSearchEngine.cs
+++ b/CoreWiki.Application/Articles/Search/Impl/ArticlesDbSearchEngine.cs
@@ -24,7 +24,7 @@ namespace CoreWiki.Application.Articles.Search.Impl
 			var filteredQuery = query.Trim();
 			var offset = (pageNumber - 1) * resultsPerPage;
 
-			var (articles, totalFound) = _articleRepo.GetArticlesForSearchQuery(filteredQuery, offset, resultsPerPage);
+			var (articles, totalFound) = await _articleRepo.GetArticlesForSearchQuery(filteredQuery, offset, resultsPerPage);
 
 			var result = new SearchResult<Article>
 			{

--- a/CoreWiki.Data.Abstractions/Interfaces/IArticleRepository.cs
+++ b/CoreWiki.Data.Abstractions/Interfaces/IArticleRepository.cs
@@ -18,7 +18,7 @@ namespace CoreWiki.Data.Abstractions.Interfaces
 
 		Task<Article> CreateArticleAndHistory(Article article);
 
-		(IEnumerable<Article> articles, int totalFound) GetArticlesForSearchQuery(string filteredQuery, int offset, int resultsPerPage);
+		Task<(IEnumerable<Article> articles, int totalFound)> GetArticlesForSearchQuery(string filteredQuery, int offset, int resultsPerPage);
 
 		Task<bool> IsTopicAvailable(string articleSlug, int articleId);
 

--- a/CoreWiki.Data/Repositories/ArticleRepository.cs
+++ b/CoreWiki.Data/Repositories/ArticleRepository.cs
@@ -71,7 +71,7 @@ namespace CoreWiki.Data.EntityFramework.Repositories
 				.AnyAsync(a => a.Slug == articleSlug && a.Id != articleId);
 		}
 
-		public (IEnumerable<Article> articles, int totalFound) GetArticlesForSearchQuery(string filteredQuery, int offset, int resultsPerPage)
+		public async Task<(IEnumerable<Article> articles, int totalFound)> GetArticlesForSearchQuery(string filteredQuery, int offset, int resultsPerPage)
 		{
 			// WARNING:  This may need to be further refactored to allow for database optimized search queries
 
@@ -82,7 +82,7 @@ namespace CoreWiki.Data.EntityFramework.Repositories
 					|| a.Content.ToUpper().Contains(filteredQuery.ToUpper())
 				).Select(a => a.ToDomain());
 			var articleCount = articles.Count();
-			var list = articles.Skip(offset).Take(resultsPerPage).OrderByDescending(a => a.ViewCount).ToList();
+			var list = await articles.Skip(offset).Take(resultsPerPage).OrderByDescending(a => a.ViewCount).ToListAsync();
 
 			return (list, articleCount);
 		}

--- a/CoreWiki/Pages/CreateArticleFromLink.cshtml.cs
+++ b/CoreWiki/Pages/CreateArticleFromLink.cshtml.cs
@@ -71,7 +71,7 @@ namespace CoreWiki.Pages
 				taskList.Add(_mediator.Send(createCmd));
 			});
 
-			Task.WaitAll(taskList.ToArray());
+			await Task.WhenAll(taskList.ToArray());
 
 			return Redirect($"/wiki/{(slug == UrlHelpers.HomePageSlug ? "" : slug)}");
 		}

